x-pack/plugins/lens/public/xy_visualization/xy_config_panel/threshold_panel.tsx

```diff
@@ -109,13 +109,82 @@ const IconSelect = ({
   );
 };
 
+function getIconPositionOptions({
+  isHorizontal,
+  axisMode,
+}: {
+  isHorizontal: boolean;
+  axisMode: YConfig['axisMode'];
+}) {
+  const options = [
+    {
+      id: `${idPrefix}auto`,
+      label: i18n.translate('xpack.lens.xyChart.thresholdMarker.auto', {
+        defaultMessage: 'Auto',
+      }),
+      'data-test-subj': 'lnsXY_markerPosition_auto',
+    },
+  ];
+  const topLabel = i18n.translate('xpack.lens.xyChart.markerPosition.above', {
+    defaultMessage: 'Top',
+  });
+  const bottomLabel = i18n.translate('xpack.lens.xyChart.markerPosition.below', {
+    defaultMessage: 'Bottom',
+  });
+  const leftLabel = i18n.translate('xpack.lens.xyChart.markerPosition.left', {
+    defaultMessage: 'Left',
+  });
+  const rightLabel = i18n.translate('xpack.lens.xyChart.markerPosition.right', {
+    defaultMessage: 'Right',
+  });
+  if (axisMode === 'bottom') {
+    const bottomOptions = [
+      {
+        id: `${idPrefix}above`,
+        label: isHorizontal ? rightLabel : topLabel,
+        'data-test-subj': 'lnsXY_markerPosition_above',
+      },
+      {
+        id: `${idPrefix}below`,
+        label: isHorizontal ? leftLabel : bottomLabel,
+        'data-test-subj': 'lnsXY_markerPosition_below',
+      },
+    ];
+    if (isHorizontal) {
+      // above -> below
```

Comment:
these comments are unclear here for me, could you explain?

Reply:
The `reverse` is executed to keep the order in the comments for the button groups.
While we want to show `Auto | Left | Right` in Vertical mode, when in Horizontal mode the order would be best as `Auto | Top | Bottom` in my opinion.
Those comments are a not so clear way to express this logic 😅